### PR TITLE
[cli] [tabletconn] Migrate `go/vt/vttablet/tabletconn` to `pflag`

### LINKED ISSUE #10978: Switch flag definitions to be on pflag instead of flag in `package go/vt/vttablet/tabletconn`
Part of https://github.com/vitessio/vitess/issues/10697.

Current flags:

```
$ git grep -E "\bflag\.[A-Z]" -- go/vt/vttablet/tabletconn/*.go
go/vt/vttablet/tabletconn/tablet_conn.go:       TabletProtocol = flag.String("tablet_protocol", "grpc", "how to talk to the vttablets")
```

### PR DESCRIPTION
**Pull request by ajm188**
## Description

Title says it all.

Here's my analysis for the commands included:

```
vitess.io/vitess/go/cmd/rulesctl => nope
    vitess.io/vitess/go/cmd/rulesctl/cmd
    vitess.io/vitess/go/cmd/rulesctl/common
vitess.io/vitess/go/cmd/vtadmin => nope
vitess.io/vitess/go/cmd/vtbench => nope
vitess.io/vitess/go/cmd/vtcombo => yep
vitess.io/vitess/go/cmd/vtctl => yep
vitess.io/vitess/go/cmd/vtctld => yep
vitess.io/vitess/go/cmd/vtctldclient => yes (only when using local shim, which _actually_ happens via `go/cmd/vtctl`)
    vitess.io/vitess/go/cmd/vtctldclient/command
    vitess.io/vitess/go/cmd/vtctldclient/docgen
vitess.io/vitess/go/cmd/vtexplain => nope (was unsure, but tests pass without it)
vitess.io/vitess/go/cmd/vtgate => yep
vitess.io/vitess/go/cmd/vtgateclienttest => nope (was unsure, but tests pass without it)
vitess.io/vitess/go/cmd/vttablet => yep (for streaming from other tablets)
vitess.io/vitess/go/cmd/zk => nope (imports `vtctl` for `DecodeContent` only)
```

## Related Issue(s)

Closes #10978.

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/vttablet/tabletconn/tablet_conn.go [hunk 1]
```diff
@@ -36,9 +38,31 @@ var (
 
 var (
 	// TabletProtocol is exported for unit tests
-	TabletProtocol = flag.String("tablet_protocol", "grpc", "how to talk to the vttablets")
+	//
+	// (TODO|andrew): rewrite tests so this does not need to be exported.
+	TabletProtocol = "grpc"
```
Comment (ajm188) [inline go/vt/vttablet/tabletconn/tablet_conn.go:43]: While debugging the fix in `healthcheck_test`, I found that the following change will let us un-export `TabletProtocol`.

- Pros: let's us protect the global variable, and probably reduce the chance of accidentally writing racy tests.
- Cons: more verbose

```diff
--- a/go/vt/vttablet/tabletconn/tablet_conn.go
+++ b/go/vt/vttablet/tabletconn/tablet_conn.go
- var TabletProtocol = "grpc"
+ var tabletProtocol = "grpc"

- func registerFlags(fs *pflag.FlagSet) {
-    fs.StringVar(&TabletProtocol, "tablet_protocol", "...")
+   fs.StringVar(&tabletProtocol, "tablet_protocol", "...")
}

--- a/go/vt/discovery/healthcheck_test.go
+++ b/go/vt/discovery/healthcheck_test.go
+const testName = "go.vt.discovery.test"
+
 func init() {
        tabletconn.RegisterDialer("fake_gateway", tabletDialer)
 
+       servenv.OnParseFor(testName, tabletconn.RegisterFlags)
+       tmp := os.Args[:]
+       os.Args = []string{testName}
+       servenv.ParseFlags(testName)
+       os.Args = tmp
+
        // log error
-       if err := flag.Set("tablet_protocol", "fake_gateway"); err != nil {
+       if err := pflag.Set("tablet_protocol", "fake_gateway"); err != nil {
                log.Errorf("failed to set flag \"tablet_protocol\" to \"fake_gateway\":%v", err)
        }
```

thoughts? I can do this everywhere if we prefer this over the "export the variable" approach we currently use (cc @deepthi )

**Review by deepthi — APPROVED** [review]
I reviewed the decisions, they look good.

File: go/vt/vttablet/tabletconn/tablet_conn.go (hunk 1 above)
Comment (deepthi) [reply go/vt/vttablet/tabletconn/tablet_conn.go:43]: I vote yes. More test code is better than exporting a variable that we don't really want to export.

Comment (ajm188) [reply go/vt/vttablet/tabletconn/tablet_conn.go:43]: https://github.com/vitessio/vitess/issues/11009

**Pull request merged**